### PR TITLE
Architecture overview

### PR DESCRIPTION
**Pull request by kaapstorm**
Adds architecture overview to documentation for self-hosting. Based on [this presentation](https://docs.google.com/presentation/d/1wR13WMgVXpT_tzXJLSgIL8l3sxELfAyQeAikX7IT39Q/edit#slide=id.g35251bddd4_0_171).

Easier to review [the rendered document](https://github.com/dimagi/commcare-cloud/blob/nh/architecture/hosting_docs/source/installation/prereqs/2-architecture.rst).

Note: The base branch is **sr/docs**

##### ENVIRONMENTS AFFECTED
None. Documentation only.

### CODE REVIEW

File: hosting_docs/source/installation/prereqs/2-architecture.rst [hunk 1]
```diff
@@ -1,4 +1,284 @@
...
+* "create a new domain" is a request that creates a "domain" document
+  type (in Couch).
+* "invite new web user" is a request that creates a "web user" document
+  type.
+
+
+BlobDB
+------
+
+Riak is a key-value store. BlobDB is the wrapper we've written around
```
Comment (millerdev) [inline hosting_docs/source/installation/prereqs/2-architecture.rst:78]: We haven't used Riak for quite some time. I think the recommended backend is now [Minio](https://docs.min.io/). References should be changed below as well.

**Comment by kaapstorm** [comment]
@Charl1996 @sravfeyn @proteusvacuum I've dropped outdated references to Riak, and merged upstream changes.

**Review by Charl1996 — APPROVED** [review]

**Comment by sravfeyn** [comment]
Hey @kaapstorm, wondering if you weren't able to see my comment.
> We don't need to go into details of the HQ architecture here. This is already documented at https://commcare-hq.readthedocs.io/overview/architecture.html we just need to give a very high level overview and link to https://commcare-hq.readthedocs.io/overview/architecture.html from here for more info. Maintaining docs on same topic on two different places is not useful.

**Comment by kaapstorm** [comment]
Oh dear. Sorry, I did miss that. And I totally agree that we shouldn't maintain two documents on the same topic.

There is a lot of overlap, but there are bits in each doc that aren't in the other. Do you reckon I should merge them, or just revert this one?

:thinking: I'm leaning towards merging this one into .../overview/architecture.html.

**Comment by sravfeyn** [comment]
No worries!

> Do you reckon I should merge them, or just revert this one?

There main reason I think [HQ Docs](https://commcare-hq.readthedocs.io/overview/architecture.html ) should be updated over maintaining an architecture doc in commcare-cloud.readthedocs.io is that the [HQ Docs](https://commcare-hq.readthedocs.io/overview/architecture.html) documents django/application specific details where the architecture doc fits nicely, an HQ dev may not need all commcare-cloud docs but will definitely need the architecture doc. So keeping the architecture doc in HQ docs up to date might be better.

We could update the HQ docs architecture with whatever new changes you wrote here and in commcare-cloud docs, just link to that with a short blurb instead of maintaining one here as well.

I am happy to do it if you don't have chance.

**Pull request merged**